src/scoring/scoring.service.ts

```diff
@@ -185,63 +185,55 @@ export class ScoringService {
   calculateScores(PDAs: Array<IssuedPDA>): PDAScores<ScoringDomainBlock> {
     const scoresOutput: PDAScores<ScoringDomainBlock> = {};
     const GIDToEthVotingAddr: Record<string, string> = {};
-    const canBeScoredPDAs: Array<IssuedPDA> = [];
 
-    // find voting address and PDAs that can be scored
+    // Find voting address and PDAs that can be scored
     for (let index = 0; index < PDAs.length; index++) {
       const PDA = PDAs[index];
 
       const GATEWAY_ID = PDA.dataAsset.owner.gatewayId;
-      // const PDA_TYPE = PDA.dataAsset.claim.pdaType;
-
-      const PDA_check: any = lodash.find(PDAs, (PDA_check) => {
-        return (
-          PDA_check.dataAsset.claim.pdaType === 'citizen' &&
-          PDA_check.dataAsset.claim.pdaSubtype === 'POKT DAO' &&
-          PDA_check.dataAsset.owner.gatewayId === PDA.dataAsset.owner.gatewayId
-        );
-      });
-      if (PDA_check && !(GATEWAY_ID in GIDToEthVotingAddr)) {
-        GIDToEthVotingAddr[GATEWAY_ID] =
-          PDA_check.dataAsset.claim.votingAddress;
-      }
+      const PDA_TYPE = PDA.dataAsset.claim.pdaType;
 
-      if (GATEWAY_ID in GIDToEthVotingAddr) {
-        canBeScoredPDAs.push(PDA);
+      if (
+        PDA_TYPE === 'citizen' &&
+        PDA.dataAsset.claim.pdaSubtype === 'POKT DAO'
+      ) {
+        GIDToEthVotingAddr[GATEWAY_ID] = PDA.dataAsset.claim.votingAddress;
       }
     }
 
-    // calculate scores of PDAs that can be scored
-    for (let index = 0; index < canBeScoredPDAs.length; index++) {
-      const PDA = canBeScoredPDAs[index];
+    // Calculate scores of PDAs that can be scored
+    for (let index = 0; index < PDAs.length; index++) {
+      const PDA = PDAs[index];
 
       const GATEWAY_ID = PDA.dataAsset.owner.gatewayId;
       const PDA_TYPE = PDA.dataAsset.claim.pdaType;
       const ETH_VOTING_ADDR = GIDToEthVotingAddr[GATEWAY_ID];
```

Comment:
Please move this line inside the next `if` block for better clarification and to avoid unexpected issues. The value of this expression is `undefined` if the `GATEWAY_ID` does not exist in the `GIDToEthVotingAddr` object.